autotx/agents/TokenResearchAgent.py

```diff
@@ -0,0 +1,173 @@
+                ],
+                "price_change_percentage_1y": token_information["market_data"][
+                    "price_change_percentage_1y"
+                ],
+            }
+        )
+
+class GetAvailableCategories(BaseTool):
+    name: str = "get_available_categories"
+    description: str = dedent(
+        """
+        Retrieve all categories
+        """
+    )
+
+    def _run(self):
+        get_categories = "/coins/categories/list"
+        categories = coingecko.request(endpoint=get_categories)
+        return json.dumps(categories)
+
+
+class GetTokensBasedOnCategory(BaseTool):
+    name: str = "get_tokens_based_on_category"
+    description: str = dedent(
+        """
+        Retrieve all tokens from a given category
+
+        Args:
+            category (str): Category to retrieve tokens
+            sort_by (str): Sort tokens by field. It can be: "volume_desc" | "volume_asc" | "market_cap_desc" | "market_cap_asc"
+                "market_cap_desc" should be the default if none is defined
+            limit (int): The number of tokens to return from the category
+        """
+    )
+
+    def _run(self, category: str, sort_by: str, limit: int):
+        get_tokens_endpoint = f"/coins/markets?vs_currency=usd&category={category}&order={sort_by}&price_change_percentage=1h,24h,7d,14d,30d,200d,1y"
+        tokens_in_category = coingecko.request(endpoint=get_tokens_endpoint)
+
+        tokens = [
+            {
+                "id": token["id"],
+                "symbol": token["symbol"],
+                "market_cap": token["market_cap"],
+                "total_volume_last_24h": token["total_volume"],
+                "price_change_percentage_1h": token[
+                    "price_change_percentage_1h_in_currency"
+                ],
+                "price_change_percentage_24h": token[
+                    "price_change_percentage_24h_in_currency"
+                ],
+                "price_change_percentage_7d": token[
+                    "price_change_percentage_7d_in_currency"
+                ],
+                "price_change_percentage_14d": token[
+                    "price_change_percentage_14d_in_currency"
+                ],
+                "price_change_percentage_30d": token[
+                    "price_change_percentage_30d_in_currency"
+                ],
+                "price_change_percentage_200d": token[
+                    "price_change_percentage_200d_in_currency"
+                ],
+                "price_change_percentage_1y": token[
+                    "price_change_percentage_1y_in_currency"
+                ],
+            }
+            for token in tokens_in_category[:limit]
+        ]
+
+        return json.dumps(tokens)
+
+
+class TokenResearchAgent(AutoTxAgent):
+    def __init__(self):
+        super().__init__(
+            name="token-researcher",
+            role="Highly specialized AI assistant with expertise in cryptocurrency analysis and investment strategies",
```

Comment:
I think these prompts should avoid using the word "investment" or "strategy" and instead simply talk about the analytical utilities it provides.

For example, I would change `with expertise in cryptocurrency analysis and investment strategies` to `with expertise in researching cryptocurrencies and analyzing the market`